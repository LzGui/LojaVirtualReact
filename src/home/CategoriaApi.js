const URL_BASE = 'https://lojavirtual-api--tarleylana.repl.co/api/v1/categorias'

const getAll = () => fetch(URL_BASE, { method: 'GET' })
  .then(res => res.json())
  .catch(err => console.error('Falha ao tentar obter categorias da base', err.message))

export {
  getAll
}