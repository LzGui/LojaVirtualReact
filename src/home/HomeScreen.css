section {
  display: flex;
}

section article {
  flex-basis: 40%;
  background: linear-gradient(aquamarine);
  padding: 15px;
}

section aside {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
}

@media (max-width: 900px) {
  body {
    width: auto;
  }
  section {
    display: flex;
    flex-direction: column;
    flex-basis: 200px;
  }
}