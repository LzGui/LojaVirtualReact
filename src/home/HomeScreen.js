import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import Header from './Header'
import Footer from './Footer'
import TextSite from './TextSite'
import CardCategoria from './CardCategoria'
import './HomeScreen.css'
import * as CategoriaApi from './CategoriaApi'

/*import appleImg from '../img/apple.png'
import playstationImg from '../img/playstation.png'
import windowsImg from '../img/windows.png'
import androidImg from '../img/android.png'*/

export default () => {

  const [listaCategorias, setListaCategorias] = useState([])

  useEffect(() => {
    carregarCategorias()
  }, []) //arrays vazio vai executar somente 1 vez

  const carregarCategorias = async () => {
    const categorias = await CategoriaApi.getAll()
    setListaCategorias(categorias)
  }

  return (
    <div data-testid='home-screen'>
      <Header />
      <section>
        <article>
          <TextSite />
        </article>
        <aside>
          {
            listaCategorias.map(categ =>
              <Link to={'/categorias/' + categ.id}>
                <CardCategoria
                  nome={categ.nome}
                  img={categ.img128}
                />
              </Link>)
          }

        </aside>
      </section>
      <Footer />
    </div>
  )
}