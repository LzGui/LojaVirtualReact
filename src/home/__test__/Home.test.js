import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render, wait } from '@testing-library/react'
import HomeScreen from '../HomeScreen'
import * as CategoriaApi from '../CategoriaApi'

test('Validar fotografia da tela', () => {
    const { getByTestId } = render(<HomeScreen />)
    const div = getByTestId('home-screen')
    expect(div).toMatchSnapshot()
})

test('API devolvendo 1 categoria', async () => {
    CategoriaApi.getAll = jest.fn(() => {
        return [{
            id: 1,
            nome: 'Android Teste',
            ativa: true,
            img64: 'http://img64.png',
            img128: 'http://img128.png',
            img256: 'http://img256.png',
        }]
    })

    const { getByText } = render(<HomeScreen />, { wrapper: MemoryRouter })
    await wait(() => getByText('Android Teste'))

    expect(getByText('Android Teste')).toBeInTheDocument()
})