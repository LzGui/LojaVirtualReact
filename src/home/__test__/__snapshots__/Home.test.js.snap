// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar fotografia da tela 1`] = `
<div
  data-testid="home-screen"
>
  <header
    data-testid="main-header"
  >
    <img
      alt="Logo"
      src="logo.png"
    />
    <h1>
      Loja Virtual
    </h1>
  </header>
  <section>
    <article>
      <p
        class="TextIntro"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Unde in enim non voluptas, ducimus tenetur praesentium sequi et ut, laboriosam autem dolores reiciendis inventore atque eum. Libero placeat rerum dolor!
      </p>
    </article>
    <aside />
  </section>
  <footer>
    Â© COPYRIGHT 2021, INSTITUTO CULTURAL NEWTON PAIVA FERREIRA LTDA.
  </footer>
</div>
`;
